netcore/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -163,10 +163,17 @@ public void DisableComObjectEagerCleanup ()
 			// no-op
 		}
 
+		[MethodImplAttribute(MethodImplOptions.InternalCall)]
+		extern static int GetCurrentProcessorNumber ();		
+
 		public static int GetCurrentProcessorId ()
 		{
-			// TODO: Implement correctly
-			return Environment.CurrentManagedThreadId;
+			int id = GetCurrentProcessorNumber ();
+
+			if (id < 0)
+				id = Environment.CurrentManagedThreadId;
```

Comment:
This logic is from 
https://github.com/dotnet/runtime/blob/aedf8f52006619ef5d4eca65d79f42cc4b7bc402/src/coreclr/src/System.Private.CoreLib/src/System/Threading/ProcessorIdCache.cs#L30-L35